miio/gateway_scripts.py

```diff
@@ -0,0 +1,332 @@
+    return dumps(
+        _inflate(
+            "taptap",
+            "[1,18,2,85,[6,512],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "tap_twice",
+        )
+    )
+
+
+def build_shakeair(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_cube.v1",
+    message_id=0,
+):
+
+    return dumps(
+        _inflate(
+            "shakeair",
+            "[1,18,2,85,[0,0],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "shake_air",
+        )
+    )
+
+
+def build_rotate(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_cube.v1",
+    message_id=0,
+):
+
+    return dumps(
+        _inflate(
+            "rotate",
+            "[1,12,3,85,[1,0],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "rotate",
+            "[1,19,7,1006,[42,[6066005667474548,12,3,85,0]],0,0]",
+        )
+    )
+
+
+def build_singlepress(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_switch.aq3",
+    message_id=0,
+):
+
+    return dumps(
+        _inflate(
+            "singlepress",
+            "[1,13,1,85,[0,1],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "click",
+        )
+    )
+
+
+def build_doublepress(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_switch.aq3",
+    message_id=0,
+):
+
+    return dumps(
+        _inflate(
+            "doublepress",
+            "[1,13,1,85,[0,2],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "double_click",
+        )
+    )
+
+
+def build_longpress(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_switch.aq3",
+    message_id=0,
+):
+
+    return dumps(
+        _inflate(
+            "longpress",
+            "[1,13,1,85,[0,16],0,0]",
+            source_sid,
+            source_model,
+            target_id,
+            target_ip,
+            target_model,
+            message_id,
+            "long_click_press",
+        )
+    )
+
+
+def build_shake(
+    source_sid,
+    target_ip,
+    target_model=fake_device_model,
+    target_id=fake_device_id,
+    source_model="lumi.sensor_switch.aq3",
```

Comment:
maybe consider adding "lumi.sensor_switch.aq3" as a property of the subdevice class:
```
    @property
    def model(self):
        """Return the model of the device."""
        return "lumi.sensor_switch.aq3"
```
also include a default for the base class:
```
    @property
    def model(self):
        """Return the model of the device."""
        return "lumi.unknown"
```

I think this can be usefull for more functions in the future and this can be used to show in HomeAssistant as the model in the device info in the GUI.

That property can than be passed to these functions.

Reply:
Probably someone will eventually figure out how to get the model directly from the gateway, I have tryed but not suceeded yet since I cannot yet capture packets.
I know the models are shown in the MiHome app if you go to the gateway device --> click ... --> about --> Hub info.
That shows a list containing dicts that have model, sid, and name, would be very usefull to get that list since we would then also have nice default names for inside HomeAsssitant while setting up the integration